### PR TITLE
`String#undump`

### PR DESCRIPTION
**Pull request by bigH**
Adds [`undump`](https://ruby-doc.org/core-2.6.4/String.html#method-i-undump
) to the `String` rbi file.

### Motivation
[`undump`](https://ruby-doc.org/core-2.6.4/String.html#method-i-undump
)

### Test plan
NA

### CODE REVIEW

File: rbi/core/string.rbi [hunk 1]
```diff
@@ -2066,4 +2066,7 @@ class String < Object
     .returns(T.nilable(String))
   end
   def slice(arg0, arg1=T.unsafe(nil)); end
+
```
Comment (DarkDimius) [inline rbi/core/string.rbi]: could you add a doc for this method?

Comment (bigH) [reply rbi/core/string.rbi]: sure

**Comment by bigH** [comment]
@DarkDimius ptal

File: rbi/core/string.rbi [hunk 2]
```diff
@@ -2066,4 +2066,12 @@ class String < Object
     .returns(T.nilable(String))
   end
   def slice(arg0, arg1=T.unsafe(nil)); end
-end
```
Comment (DarkDimius) [inline rbi/core/string.rbi]: removing this `end` made the file not parse.

File: rbi/core/string.rbi [hunk 3]
```diff
@@ -2066,4 +2066,12 @@ class String < Object
...
+
+# Produces unescaped version of str. See also #dump because #undump does
+  # inverse of #dump.
+  #
+  # ```ruby
+  # "\"hello \\n ''\"".undump #=> "hello \n ''"
+  # ```
+  sig {returns(String)}
+  def undump; end
```
Comment (DarkDimius) [inline rbi/core/string.rbi]:
```suggestion
  def undump; end
end
```

**Comment by DarkDimius** [comment]
Closing it as stale, please feel free to reopen.

**Pull request closed without merging**